speech.js

```diff
@@ -134,40 +186,183 @@ class Speech {
     this.apiKey = apiKey;
   }
 
-  async fetchVoices() {
-    return fetch(URL_VOICES, {
+  async fetchVoices(options = {}) {
+    const starred = options.starred || false;
+    const owner = options.owner || 'all';
+    let url = `${_BASE_URL}${_LIST_VOICES_ENDPOINT}?starred=${starred}&owner=${owner}`;
+    return fetch(url, {
+      headers: this._getHeaders(),
+      method: "GET",
+    })
+      .then(response => {
+        return this._handle_response_errors(response)
+          .then(() => response.json());
+      })
+      .catch((error) => {
+        throw new Error(`[fetchVoices] ${error.message}`);
+      });
+  }
+
+  async fetchVoice(voice) {
+    const url = `${_BASE_URL}${_VOICE_ENDPOINT.replace('{id}', voice)}`;
+    return fetch(url, {
       headers: this._getHeaders(),
-      method: 'GET'
-    }).then(response => response.json());
+      method: "GET",
+    })
+      .then(response => {
+        return this._handle_response_errors(response)
+          .then(() => response.json());
+      })
+      .catch((error) => {
+        throw new Error(`[fetchVoice] ${error.message}`);
+      });
   }
 
+  async createVoice(name, enhance, filenames, type = 'instant', gender = null, description = null) {
+    if (name.length === 0) {
+      throw new Error('[Speech.createVoice] Name must be non-empty.');
+    }
+    if (filenames.length === 0) {
+      throw new Error('[Speech.createVoice] Filenames must be non-empty.');
+    }
+
+    const metadata = JSON.stringify({
+      name,
+      enhance,
+      type,
+      gender,
+      description,
+    });
+
+    const formData = new FormData();
+    formData.append('metadata', metadata, {
+      contentType: 'application/json',
+    });
+    filenames.forEach((filename) => {
+      formData.append('file_field', fs.createReadStream(filename));
+    }
+    );
+    return fetch(`${_BASE_URL}${_CREATE_VOICE_ENDPOINT}`, {
```

Comment:
I think these should be using the await pattern, since we're in async/await land

something like:

try {
const response = await fetch();
this._handle_response_errors(response);
return response.json();
} catch (error) {
  // throw error
}